src/setups/extrude_geometry.jl

```diff
@@ -0,0 +1,212 @@
+"""
+    ExtrudeGeometry(geometry; particle_spacing, direction, n_extrude=1,
+                    velocity=zeros(length(direction)),
+                    mass=nothing, density=nothing, pressure=0.0)
+Extrude either a line, a plane or a shape along a specific direction.
+
+# Arguments
+- `geometry`:           Either points defining a 3D plane (2D line)
+                        or particle coordinates defining a specific shape.
+
+# Keywords
+- `particle_spacing`:   Spacing between the particles.
+- `direction`:          Vector defining the extrusion direction.
+- `n_extrude=1`         Number of `geometry` layers in extrude direction.
+- `velocity`:           Either a function mapping each particle's coordinates to its velocity,
+                        or, for a constant fluid velocity, a vector holding this velocity.
+                        Velocity is constant zero by default.
+- `mass`:               Either `nothing` (default) to automatically compute particle mass from particle
+                        density and spacing, or a function mapping each particle's coordinates to its mass,
+                        or a scalar for a constant mass over all particles.
+- `density`:            Either a function mapping each particle's coordinates to its density,
+                        or a scalar for a constant density over all particles.
+                        Obligatory when not using a state equation. Cannot be used together with
+                        `state_equation`.
+- `pressure`:           Scalar to set the pressure of all particles to this value.
+                        This is only used by the [`EntropicallyDampedSPHSystem`](@ref) and
+                        will be overwritten when using an initial pressure function in the system.
+                        Cannot be used together with hydrostatic pressure gradient.
+- `tlsph`:              With the [`TotalLagrangianSPHSystem`](@ref), particles need to be placed
+                        on the boundary of the shape and not one particle radius away, as for fluids.
+                        When `tlsph=true`, particles will be placed on the boundary of the shape.
+
+# Examples
+```julia
+# 2D
+p1 = [0.0, 0.0]
+p2 = [1.0, 1.0]
+
+direction = [-1.0, 1.0]
+
+shape = ExtrudeGeometry((p1, p2); direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+
+# 3D Plane
+p1 = [0.0, 0.0, 0.0]
+p2 = [0.5, 1.0, 0.0]
+p3 = [1.0, 0.2, 0.0]
+
+direction = [0.0, 0.0, 1.0]
+
+shape = ExtrudeGeometry((p1, p2, p3); direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+
+# Extrude a 2D shape to a 3D shape
+shape = SphereShape(0.1, 0.5, (0.2, 0.4), 1000.0, n_layers=3,
+                    sphere_type=RoundSphere(end_angle=pi))
+
+direction = [0.0, 0.0, 1.0]
+
+shape = ExtrudeGeometry(shape; direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+```
+
+!!! warning
+    `particle_spacing` between extrusion layers may differ from shapes `particle_spacing`.
```

Comment:
And can the `density` of an `InitialCondition` just be extruded as well? This would be useful for creating a 3D parallelogram tank, and it would make the API easier by not having to pass `density` all the time.

Reply:
Speaking of API: After our discussion 2 days ago, it would be good to add a warning
```suggestion
!!! warning "Experimental Implementation"
    This is an experimental feature and may change in any future releases.
```
so that we can release breaking changes to it without releasing a new major release.
We should do that with all new features, and remove the warning once we've used it for a while and we're happy with the API.